composite/outlier-detection/src/source.ts

```diff
@@ -0,0 +1,48 @@
+import { Execute } from '@chainlink/types'
+import * as GenesisVolatility from '@chainlink/genesis-volatility-adapter'
+import * as XBTO from '@chainlink/xbto-adapter'
+
+export enum SourceDataProvider {
+  XBTO = 'xbto',
+  GenesisVolatility = 'genesisvolatility',
+}
+
+const isSourceDataProvider = (envVar?: string): envVar is SourceDataProvider =>
+  Object.values(SourceDataProvider).includes(envVar as any)
+
+export const getSourceDataProviders = (): SourceDataProvider[] => {
+  const sourceDataProviders = (process.env.SOURCE_ADAPTERS || '').split(',')
+  const adapters: SourceDataProvider[] = []
+
+  sourceDataProviders.forEach((provider) => {
+    if (isSourceDataProvider(provider)) adapters.push(provider as SourceDataProvider)
+  })
+
+  return adapters
```

Comment:
This can be just a filter of the original array:

```suggestion
  return sourceDataProviders.filter(isSourceDataProvider)
    .map(provider => provider as SourceDataProvider)
```